packages/python/plotly/plotly/tests/test_optional/test_px/test_px_input.py

```diff
@@ -138,6 +140,52 @@ def test_name_heuristics():
     assert fig.data[0].hovertemplate == "y=%{marker.size}<br>x=%{y}<extra></extra>"
 
 
+def test_performance_b64():
+    np_arr_1 = np.random.random(1000000)
+    np_arr_2 = np.random.random(1000000)
+
+    # Test the performance of the base64 arrays
+    np_arr_start = time.time()
+    fig = px.scatter(x=np_arr_1, y=np_arr_2, width=800, height=800)
+    fig.update_layout(margin=dict(l=0, r=0, t=0, b=0))
+    np_arr_time_elapsed = time.time() - np_arr_start
+
+    # Test the performance of the raw arrays
+    list_start = time.time()
+    fig = px.scatter(x=np_arr_1.tolist(), y=np_arr_2.tolist(), width=800, height=800)
+    fig.update_layout(margin=dict(l=0, r=0, t=0, b=0))
+    list_time_elapsed = time.time() - list_start
+
+    # b64 should be faster than raw
+    assert (np_arr_time_elapsed / list_time_elapsed) < 0.7
```

Comment:
Let's add `test_performance_b64_float32` as well as for size which could demonstrate similar/better results.
Thank you!